autoload/go/list.vim

```diff
@@ -22,50 +31,96 @@ function! go#list#Window(...)
         endif
     endif
 
-    exe 'lopen '. height
+    if l:quickfix == 0
+      exe 'lopen ' . height
+    else
+      exe 'copen ' . height
+    endif
 endfunction
 
 
 " Get returns the current list of items from the location list
-function! go#list#Get()
-  return getloclist(0)
+function! go#list#Get(quickfix)
+    let l:quickfix = go#list#Type(a:quickfix)
+    if l:quickfix == 0
+        return getloclist(0)
+    else
+        return getqflist()
+    endif
 endfunction
 
 " Populate populate the location list with the given items
-function! go#list#Populate(items)
-	call setloclist(0, a:items, 'r')
+function! go#list#Populate(quickfix, items)
+    let l:quickfix = go#list#Type(a:quickfix)
+    if l:quickfix == 0
+        call setloclist(0, a:items, 'r')
+    else
+        call setqflist(a:items, 'r')
+    endif
 endfunction
 
 function! go#list#PopulateWin(winnr, items)
-	call setloclist(a:winnr, a:items, 'r')
+    call setloclist(a:winnr, a:items, 'r')
 endfunction
 
 " Parse parses the given items based on the specified errorformat nad
 " populates the location list.
-function! go#list#ParseFormat(errformat, items)
-  " backup users errorformat, will be restored once we are finished
-  let old_errorformat = &errorformat
+function! go#list#ParseFormat(quickfix, errformat, items)
+    let l:quickfix = go#list#Type(a:quickfix)
+    " backup users errorformat, will be restored once we are finished
+    let old_errorformat = &errorformat
 
-	" parse and populate the location list
-  let &errorformat = a:errformat
-	lgetexpr a:items
+    " parse and populate the location list
+    let &errorformat = a:errformat
+    if l:quickfix == 0
+        lgetexpr a:items
+    else
+        cgetexpr a:items
+    endif
 
-	"restore back
-  let &errorformat = old_errorformat
+    "restore back
+    let &errorformat = old_errorformat
 endfunction
 
-" Parse parses the given items based on the global errorformat nad
+" Parse parses the given items based on the global errorformat and
 " populates the location list.
-function! go#list#Parse(items)
-	lgetexpr a:items
+function! go#list#Parse(quickfix, items)
+    let l:quickfix = go#list#Type(a:quickfix)
+    if l:quickfix == 0
+        lgetexpr a:items
+    else
+        cgetexpr a:items
+    endif
 endfunction
 
 " JumpToFirst jumps to the first item in the location list
-function! go#list#JumpToFirst()
-  ll 1 
+function! go#list#JumpToFirst(quickfix)
+    let l:quickfix = go#list#Type(a:quickfix)
+    if l:quickfix == 0
+        ll 1
+    else
+        cc 1
+    endif
 endfunction
 
 " Clean cleans the location list
-function! go#list#Clean()
-	lex []
+function! go#list#Clean(quickfix)
+    let l:quickfix = go#list#Type(a:quickfix)
+    if l:quickfix == 0
+        lex []
+    else
+        cex []
+    endif
 endfunction
+
+function! go#list#Type(quickfix)
```

Comment:
I'm not sure if we need this. It's yet another abstraction and seems like we don't need it. Can you remove it? Anywhere else just let use strings. Like:

```
if a:listype == "quickfix"
endif
```

This is more readable and we don't need to manually parse what `0` or `1` means.